paper-server/patches/sources/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -2471,6 +2471,19 @@
          this.player.resetLastActionTime();
          ServerLevel serverLevel = this.player.serverLevel();
          BlockPos pos = packet.getPos();
+         if (serverLevel.hasChunkAt(pos)) {
++            // Paper start - Add API for client-side signs
++            if (!new io.papermc.paper.event.packet.UncheckedSignChangeEvent(this.player.getBukkitEntity(),
++                io.papermc.paper.util.MCUtil.toPosition(pos),
++                packet.isFrontText() ? org.bukkit.block.sign.Side.FRONT : org.bukkit.block.sign.Side.BACK,
++                filteredText.stream().map(line -> (net.kyori.adventure.text.Component) net.kyori.adventure.text.Component.text(line.raw())).toList())
```

Comment:
Idk if raw is the right thing to use here when usually the filteredOrEmpty output is passed through StringUtil.filterText.

Reply:
When I opened the original PR I was honestly stumped how FilteredText is used on the server, so I guessed I'll use the raw text the player entered. Does anyone have proper insight how this system is actually used?

Reply:
FilteredText isn't *really* used on paper at all, it's realms tech.
Raw imo is fine tho, this is "unchecked" afterall xD